lucene/queries/src/java/org/apache/lucene/queries/function/ValueSourceScorer.java

```diff
@@ -55,7 +59,13 @@ public boolean matches() throws IOException {
 
       @Override
       public float matchCost() {
-        return 100; // TODO: use cost of ValueSourceScorer.this.matches()
+        // If an external cost is set, use that
+        if (externallyMutableCost != 0.0) {
+          return externallyMutableCost;
+        }
+
+        // Cost of iteration is fixed cost + cost exposed by delegated FunctionValues instance
+        return DEF_COST + values.cost();
```

Comment:
I suppose the purpose of DEF_COST here is to add on the cost of the ValueSourceScorer.matches code that is separate from fetching the value from the FunctionValues.  That cost varies by the VSC subclass.  If we want to be thorough then maybe _this_ should be settable somehow _as well_?   One easy way to do this is to simply refactor out this one line (def cost + FV cost) into a protected method on the VSC that may be over-ridden if desired.   Or we could just say that is too pedantic, and prefer simplicity.  I'm fine either way.

Reply:
I felt that DEF_COST defines the cost of the VSC.matches call itself and if the user prefers to do a complex cost, then override matchCost() :) I agree with your approach, added the cost evaluation method as a separate class method